### DISCUSSION TITLE
How to handle escape character？

### DISCUSSION THREAD

**Discussion by OneLeal** [sourcelair/xterm.js#4649] [Q&A] [answered]
"xterm": "^5.1.0",
"xterm-addon-attach": "^0.8.0",
"xterm-addon-fit": "^0.7.0",
"xterm-addon-web-links": "^0.8.0"

when I press the key "↑" on keyboard, Xterm shows string '^[[A' rather than the last command 'ls'
![截屏2023-08-08 19 56 05](https://github.com/xtermjs/xterm.js/assets/55690294/5b829401-6874-4e40-b452-8b3ed055ebff)

**Comment by Tyriar** [comment, marked as answer]
When you press the up arrow, xterm.js sends the text-representation of the up arrow to your backend. You backend then handles that and tells xterm.js what to print to the terminal. So this question likely isn't related to xterm.js but how your backend is set up.

**Reply by OneLeal** [reply]
thanks, I got it.
